kernels/portable/cpu/util/copy_ops_util.cpp

```diff
@@ -351,6 +353,10 @@ void get_pixel_shuffle_out_target_size(
     int64_t upscale_factor,
     executorch::aten::SizesType* out_sizes,
     size_t* out_ndim) {
+  // Prevent signed integer overflow when computing upscale_factor ^ 2.
+  // This constraint is checked in check_pixel_shuffle_args.
+  ET_CHECK(upscale_factor < 32768);
```

Comment:
Nit: Can you update this to not fatal error and instead just set the runtime context state

Reply:
Yeah, I did add a non-fatal check in the pixel shuffle args check, which will in practice handle it gracefully. I added this assert here just as a sanity check, but am open to changing it if you prefer. It seemed harder to bail out at this point but I wanted to catch any UB.

Reply:
Actually it does seems simpler to just refactor the get out size calculation to be able to fail non-fatally and move the check there. I'll push another revision.